tests/Annotations/Command/QueueWatchCommandTest.php

```diff
@@ -116,6 +119,70 @@ public function it_retries_failures()
         $this->assertSame(1, $this->queue->count());
     }
 
+    /**
+     * @test
+     */
+    public function it_will_remove_queue_item_if_upsert_fails()
+    {
+        $item = new InternalSqsMessage('profile', 'username');
+        $profile = new Profile(
+            'username',
+            new PersonDetails('PreferredName', 'IndexName'),
+            new EmptySequence(),
+            new EmptySequence()
+        );
+        $data = [
+            'authority' => $this->authority,
+            'username' => 'username',
+            'email' => 'username@blackhole.elifesciences.org',
+            'display_name' => 'PreferredName',
+        ];
+        $this->transformer
+            ->expects($this->once())
+            ->method('transform')
+            ->with($item)
```

Comment:
same here for `with()`, incidentally some variables that are only used in `with()` could then be deleted for a shorter and to-the-point test